### ISSUE TITLE
bun install failed on `>=v1.0.32` while success on `v1.0.31`

### ISSUE THREAD

**Issue by arvinxx** [oven-sh/bun#9503]
### What version of Bun is running?

\>=1.0.32

### What platform is your computer?

macos/linux

### What steps can reproduce the bug?

- git clone https://github.com/lobehub/lobe-chat
- bun i

### What is the expected behavior?

install success

### What do you see instead?

install failed with 1.0.32 :
```shell
bun install v1.0.32 (5fec71bd)
  🚚 chownr... gyp info it worked if it ends with ok
gyp info using node-gyp@9.4.1
gyp info using node@18.19.0 | darwin | arm64
gyp info find Python using Python version 3.9.5 found at "/opt/homebrew/opt/python@3.9/bin/python3.9"
gyp ERR! UNCAUGHT EXCEPTION 
gyp ERR! stack Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/wenbing/CodeProjects/LobeHub/lobe-chat/node_modules/p-map/index.js from /Users/wenbing/CodeProjects/LobeHub/lobe-chat/node_modules/node-gyp/node_modules/make-fetch-happen/node_modules/cacache/lib/verify.js not supported.
gyp ERR! stack Instead change the require of index.js in /Users/wenbing/CodeProjects/LobeHub/lobe-chat/node_modules/node-gyp/node_modules/make-fetch-happen/node_modules/cacache/lib/verify.js to a dynamic import() which is available in all CommonJS modules.
gyp ERR! stack     at Object.<anonymous> (/Users/wenbing/CodeProjects/LobeHub/lobe-chat/node_modules/node-gyp/node_modules/make-fetch-happen/node_modules/cacache/lib/verify.js:5:14)
gyp ERR! stack     at Object.<anonymous> (/Users/wenbing/CodeProjects/LobeHub/lobe-chat/node_modules/node-gyp/node_modules/make-fetch-happen/node_modules/cacache/lib/index.js:6:16)
gyp ERR! stack     at Object.<anonymous> (/Users/wenbing/CodeProjects/LobeHub/lobe-chat/node_modules/node-gyp/node_modules/make-fetch-happen/lib/cache/entry.js:4:17)
gyp ERR! stack     at Object.<anonymous> (/Users/wenbing/CodeProjects/LobeHub/lobe-chat/node_modules/node-gyp/node_modules/make-fetch-happen/lib/cache/index.js:2:20)
gyp ERR! stack     at Object.<anonymous> (/Users/wenbing/CodeProjects/LobeHub/lobe-chat/node_modules/node-gyp/node_modules/make-fetch-happen/lib/fetch.js:7:15)
gyp ERR! stack     at Object.<anonymous> (/Users/wenbing/CodeProjects/LobeHub/lobe-chat/node_modules/node-gyp/node_modules/make-fetch-happen/lib/index.js:4:15)
gyp ERR! stack     at Object.<anonymous> (/Users/wenbing/CodeProjects/LobeHub/lobe-chat/node_modules/node-gyp/lib/install.js:14:15)
gyp ERR! stack     at self.commands.<computed> [as install] (/Users/wenbing/CodeProjects/LobeHub/lobe-chat/node_modules/node-gyp/lib/node-gyp.js:41:14)
gyp ERR! stack     at getNodeDir (/Users/wenbing/CodeProjects/LobeHub/lobe-chat/node_modules/node-gyp/lib/configure.js:64:20)
gyp ERR! stack     at /Users/wenbing/CodeProjects/LobeHub/lobe-chat/node_modules/node-gyp/lib/configure.js:31:7
gyp ERR! System Darwin 22.4.0
gyp ERR! command "/Users/wenbing/.nvm/versions/node/v18.19.0/bin/node" "/Users/wenbing/CodeProjects/LobeHub/lobe-chat/node_modules/.bin/node-gyp" "rebuild"
gyp ERR! cwd /Users/wenbing/CodeProjects/LobeHub/lobe-chat/node_modules/tree-sitter-yaml
gyp ERR! node -v v18.19.0
gyp ERR! node-gyp -v v9.4.1
gyp ERR! Node-gyp failed to build your package.
gyp ERR! Try to update npm and/or node-gyp and if it does not help file an issue with the package author.

error: install script from "tree-sitter-yaml" exited with 7
```

failed test actions with `bun@1.0.33` : https://github.com/lobehub/lobe-chat/actions/runs/8336796304/job/22814518710

```shell
bun install v1.0.33 (9e91e137)
 Resolving dependencies
 Resolved, downloaded and extracted [9110]
warn: incorrect peer dependency "stylelint@1[5](https://github.com/lobehub/lobe-chat/actions/runs/8336812673/job/22814702291#step:4:6).11.0"

warn: incorrect peer dependency "stylelint@15.11.0"

warn: incorrect peer dependency "stylelint@15.11.0"

warn: incorrect peer dependency "@octokit/core@5.1.0"

warn: incorrect peer dependency "@octokit/core@5.1.0"

gyp info it worked if it ends with ok
gyp info using node-gyp@9.4.1
gyp info using node@18.19.1 | linux | x64
gyp info find Python using Python version 3.10.12 found at "/usr/bin/python3"
gyp ERR! UNCAUGHT EXCEPTION 
gyp ERR! stack Error [ERR_REQUIRE_ESM]: require() of ES Module /home/runner/work/lobe-chat/lobe-chat/node_modules/p-map/index.js from /home/runner/work/lobe-chat/lobe-chat/node_modules/node-gyp/node_modules/make-fetch-happen/node_modules/cacache/lib/verify.js not supported.
gyp ERR! stack Instead change the require of index.js in /home/runner/work/lobe-chat/lobe-chat/node_modules/node-gyp/node_modules/make-fetch-happen/node_modules/cacache/lib/verify.js to a dynamic import() which is available in all CommonJS modules.
gyp ERR! stack     at Object.<anonymous> (/home/runner/work/lobe-chat/lobe-chat/node_modules/node-gyp/node_modules/make-fetch-happen/node_modules/cacache/lib/verify.js:5:14)
gyp ERR! stack     at Object.<anonymous> (/home/runner/work/lobe-chat/lobe-chat/node_modules/node-gyp/node_modules/make-fetch-happen/node_modules/cacache/lib/index.js:6:16)
gyp ERR! stack     at Object.<anonymous> (/home/runner/work/lobe-chat/lobe-chat/node_modules/node-gyp/node_modules/make-fetch-happen/lib/cache/entry.js:4:17)
gyp ERR! stack     at Object.<anonymous> (/home/runner/work/lobe-chat/lobe-chat/node_modules/node-gyp/node_modules/make-fetch-happen/lib/cache/index.js:2:20)
gyp ERR! stack     at Object.<anonymous> (/home/runner/work/lobe-chat/lobe-chat/node_modules/node-gyp/node_modules/make-fetch-happen/lib/fetch.js:7:[15](https://github.com/lobehub/lobe-chat/actions/runs/8323622199/job/22773582619#step:4:16))
gyp ERR! stack     at Object.<anonymous> (/home/runner/work/lobe-chat/lobe-chat/node_modules/node-gyp/node_modules/make-fetch-happen/lib/index.js:4:15)
gyp ERR! stack     at Object.<anonymous> (/home/runner/work/lobe-chat/lobe-chat/node_modules/node-gyp/lib/install.js:14:15)
gyp ERR! stack     at self.commands.<computed> [as install] (/home/runner/work/lobe-chat/lobe-chat/node_modules/node-gyp/lib/node-gyp.js:41:14)
gyp ERR! stack     at getNodeDir (/home/runner/work/lobe-chat/lobe-chat/node_modules/node-gyp/lib/configure.js:64:20)
gyp ERR! stack     at /home/runner/work/lobe-chat/lobe-chat/node_modules/node-gyp/lib/configure.js:31:7
gyp ERR! System Linux 6.5.0-10[16](https://github.com/lobehub/lobe-chat/actions/runs/8323622199/job/22773582619#step:4:17)-azure
gyp ERR! command "/usr/local/bin/node" "/home/runner/work/lobe-chat/lobe-chat/node_modules/.bin/node-gyp" "rebuild"
gyp ERR! cwd /home/runner/work/lobe-chat/lobe-chat/node_modules/tree-sitter-yaml
gyp ERR! node -v v[18](https://github.com/lobehub/lobe-chat/actions/runs/8323622199/job/22773582619#step:4:19).[19](https://github.com/lobehub/lobe-chat/actions/runs/8323622199/job/22773582619#step:4:20).1
gyp ERR! node-gyp -v v9.4.1
gyp ERR! Node-gyp failed to build your package.
gyp ERR! Try to update npm and/or node-gyp and if it does not help file an issue with the package author.

error: install script from "tree-sitter-yaml" exited with 7
Error: Process completed with exit code 7.
```

### Additional information

while using bun 1.0.31, install successful:

```shell
bun install v1.0.31 (e2567512)
 Resolving dependencies
 Resolved, downloaded and extracted [9134]
warn: incorrect peer dependency "stylelint@1[5](https://github.com/lobehub/lobe-chat/actions/runs/8316122799/job/22755211087#step:4:6).11.0"

warn: incorrect peer dependency "stylelint@15.11.0"

warn: incorrect peer dependency "@octokit/core@5.1.0"

warn: incorrect peer dependency "@octokit/core@5.1.0"
 Saved lockfile

 + @commitlint/cli@19.2.0
 + @ducanh2912/next-pwa@10.2.5
 + @edge-runtime/vm@3.2.0
 + @lobehub/i18n-cli@1.15.3
 + @lobehub/lint@1.22.0
 + @next/bundle-analyzer@14.1.3
 + @next/eslint-plugin-next@14.1.3
 + @peculiar/webcrypto@1.4.5
 + @testing-library/jest-dom@[6](https://github.com/lobehub/lobe-chat/actions/runs/8316122799/job/22755211087#step:4:7).4.2
 + @testing-library/react@14.2.1
 + @types/chroma-js@2.4.4
 + @types/diff@5.0.9
 + @types/json-schema@[7](https://github.com/lobehub/lobe-chat/actions/runs/8316122799/job/22755211087#step:4:8).0.15
 + @types/lodash@4.17.0
 + @types/lodash-es@4.17.12
 + @types/node@20.11.2[8](https://github.com/lobehub/lobe-chat/actions/runs/8316122799/job/22755211087#step:4:9)
 + @types/numeral@2.0.5
 + @types/react@18.2.66
 + @types/react-dom@18.2.22
 + @types/rtl-detect@1.0.3
 + @types/semver@7.5.8
 + @types/systemjs@6.13.5
 + @types/ua-parser-js@0.7.3[9](https://github.com/lobehub/lobe-chat/actions/runs/8316122799/job/22755211087#step:4:10)
 + @types/uuid@9.0.8
 + @umijs/lint@4.1.2
 + @vitest/coverage-v8@1.2.2
 + ajv-keywords@5.1.0
 + commitlint@19.2.0
 + consola@3.2.3
 + dpdm@3.14.0
 + eslint@8.57.0
 + eslint-plugin-mdx@2.3.4
 + fake-indexeddb@5.0.2
 + glob@[10](https://github.com/lobehub/lobe-chat/actions/runs/8316122799/job/22755211087#step:4:11).3.10
 + immer@10.0.4
 + jose@5.2.3
 + langfuse@3.3.1
 + langfuse-core@3.3.1
 + lodash-es@4.17.21
 + lucide-react@0.358.0
 + modern-screenshot@4.4.38
 + nanoid@5.0.6
 + next@14.1.3
 + next-auth@5.0.0-beta.15
 + next-sitemap@4.2.3
 + numeral@2.0.6
 + nuqs@1.17.1
 + ollama@0.5.0
 + openai@4.29.1
 + polished@4.3.1
 + posthog-js@1.[11](https://github.com/lobehub/lobe-chat/actions/runs/8316122799/job/22755211087#step:4:12)6.0
 + query-string@9.0.0
 + react@18.2.0
 + react-dom@18.2.0
 + react-hotkeys-hook@4.5.0
 + react-i18next@[14](https://github.com/lobehub/lobe-chat/actions/runs/8316122799/job/22755211087#step:4:15).1.0
 + react-layout-kit@1.7.4
 + react-lazy-load@4.0.1
 + react-virtuoso@4.7.2
 + react-wrap-balancer@1.1.0
 + remark@14.0.3
 + remark-gfm@3.0.1
 + remark-html@[15](https://github.com/lobehub/lobe-chat/actions/runs/8316122799/job/22755211087#step:4:16).0.2
 + rtl-detect@1.1.2
 + semver@7.6.0
 + sharp@0.33.2
 + swr@2.2.5
 + systemjs@6.14.3
 + ts-md5@1.3.1
 + ua-parser-js@1.0.37
 + url-join@5.0.0
 + use-merge-value@1.2.0
 + utility-types@3.11.0
 + uuid@9.0.1
 + yaml@2.4.1
 + zod@3.[22](https://github.com/lobehub/lobe-chat/actions/runs/8316122799/job/22755211087#step:4:23).4
 + zustand@4.5.2
 + zustand-utils@1.3.2

 [24](https://github.com/lobehub/lobe-chat/actions/runs/8316122799/job/22755211087#step:4:25)58 packages installed [[28](https://github.com/lobehub/lobe-chat/actions/runs/8316122799/job/22755211087#step:4:29).[46](https://github.com/lobehub/lobe-chat/actions/runs/8316122799/job/22755211087#step:4:47)s]

 Blocked 2 postinstalls. Run `bun pm untrusted` for details.
```

**Comment by arvinxx** [comment]
when using `bun@1.0.30`( vercel current version ) , it also sucess:
```shell
bun install v1.0.30 (1424a196)
11:19:03.571 | Resolving dependencies
11:19:15.660 | Resolved, downloaded and extracted [9124]
11:19:15.661 | warn: incorrect peer dependency "@octokit/core@5.1.0"
11:19:15.661 |  
11:19:15.661 | warn: incorrect peer dependency "stylelint@15.11.0"
11:19:15.661 |  
11:19:15.661 | warn: incorrect peer dependency "stylelint@15.11.0"
11:19:15.661 |  
11:19:15.661 | warn: incorrect peer dependency "@octokit/core@5.1.0"
11:19:15.661 |  
11:19:15.661 | warn: incorrect peer dependency "@octokit/core@5.1.0"
11:19:15.661 |  
11:19:15.661 | warn: incorrect peer dependency "@octokit/core@5.1.0"
11:19:15.768 | Saved lockfile
11:19:39.975 |  
11:19:39.975 | + @commitlint/cli@19.2.1
11:19:39.975 | + @ducanh2912/next-pwa@10.2.5
11:19:39.975 | + @edge-runtime/vm@3.2.0
11:19:39.975 | + @lobehub/i18n-cli@1.15.3
11:19:39.975 | + @lobehub/lint@1.22.0
11:19:39.975 | + @next/bundle-analyzer@14.1.3
11:19:39.975 | + @next/eslint-plugin-next@14.1.3
11:19:39.975 | + @peculiar/webcrypto@1.4.5
11:19:39.975 | + @testing-library/jest-dom@6.4.2
11:19:39.976 | + @testing-library/react@14.2.1
11:19:39.976 | + @types/chroma-js@2.4.4
11:19:39.976 | + @types/diff@5.0.9
11:19:39.976 | + @types/json-schema@7.0.15
11:19:39.976 | + @types/lodash@4.17.0
11:19:39.976 | + @types/lodash-es@4.17.12
11:19:39.976 | + @types/node@20.11.29
11:19:39.976 | + @types/numeral@2.0.5
11:19:39.976 | + @types/react@18.2.67
11:19:39.976 | + @types/react-dom@18.2.22
11:19:39.976 | + @types/rtl-detect@1.0.3
11:19:39.976 | + @types/semver@7.5.8
11:19:39.976 | + @types/systemjs@6.13.5
11:19:39.976 | + @types/ua-parser-js@0.7.39
11:19:39.976 | + @types/uuid@9.0.8
11:19:39.976 | + @umijs/lint@4.1.2
11:19:39.976 | + @vitest/coverage-v8@1.2.2
11:19:39.976 | + ajv-keywords@5.1.0
11:19:39.976 | + commitlint@19.2.1
11:19:39.976 | + consola@3.2.3
11:19:39.976 | + dpdm@3.14.0
11:19:39.977 | + eslint@8.57.0
11:19:39.977 | + eslint-plugin-mdx@2.3.4
11:19:39.977 | + fake-indexeddb@5.0.2
11:19:39.977 | + glob@10.3.10
11:19:39.977 | + husky@9.0.11
11:19:39.977 | + jsdom@24.0.0
11:19:39.977 | + just-diff@6.0.2
11:19:39.977 | + lint-staged@15.2.2
11:19:39.977 | + lodash@4.17.21
11:19:39.977 | + markdown-table@3.0.3
11:19:39.977 | + node-fetch@3.3.2
11:19:39.977 | + p-map@7.0.1
11:19:39.977 | + prettier@3.2.5
11:19:39.977 | + remark-cli@11.0.0
11:19:39.978 | + remark-parse@10.0.2
11:19:39.978 | + semantic-release@21.1.2
11:19:39.978 | + stylelint@15.11.0
11:19:39.978 | + tsx@4.7.1
11:19:39.978 | + typescript@5.4.2
11:19:39.978 | + unified@11.0.4
11:19:39.978 | + unist-util-visit@5.0.0
11:19:39.978 | + vite@5.1.6
11:19:39.978 | + vitest@1.2.2
11:19:39.978 | + vitest-canvas-mock@0.3.3
11:19:39.978 | + @ant-design/icons@5.3.4
11:19:39.978 | + @anthropic-ai/sdk@0.18.0
11:19:39.978 | + @auth/core@0.28.0
11:19:39.978 | + @aws-sdk/client-bedrock-runtime@3.535.0
11:19:39.978 | + @azure/openai@1.0.0-beta.11
11:19:39.978 | + @cfworker/json-schema@1.12.8
11:19:39.978 | + @google/generative-ai@0.2.1
11:19:39.978 | + @icons-pack/react-simple-icons@9.4.0
11:19:39.978 | + @lobehub/chat-plugin-sdk@1.32.3
11:19:39.979 | + @lobehub/chat-plugins-gateway@1.9.0
11:19:39.979 | + @lobehub/icons@1.13.1
11:19:39.979 | + @lobehub/tts@1.24.0
11:19:39.979 | + @lobehub/ui@1.135.0
11:19:39.979 | + @theguild/remark-mermaid@0.0.6
11:19:39.979 | + @vercel/analytics@1.2.2
11:19:39.979 | + ahooks@3.7.10
11:19:39.979 | + ai@3.0.12
11:19:39.979 | + antd@5.15.3
11:19:39.979 | + antd-style@3.6.1
11:19:39.979 | + brotli-wasm@2.0.1
11:19:39.979 | + chroma-js@2.4.2
11:19:39.979 | + dayjs@1.11.10
11:19:39.980 | + dexie@3.2.6
11:19:39.980 | + diff@5.2.0
11:19:39.980 | + fast-deep-equal@3.1.3
11:19:39.980 | + gpt-tokenizer@2.1.2
11:19:39.980 | + i18next@23.10.1
11:19:39.980 | + i18next-browser-languagedetector@7.2.0
11:19:39.980 | + i18next-resources-to-backend@1.2.0
11:19:39.980 | + idb-keyval@6.2.1
11:19:39.980 | + immer@10.0.4
11:19:39.980 | + jose@5.2.3
11:19:39.980 | + langfuse@3.3.1
11:19:39.980 | + langfuse-core@3.3.1
11:19:39.980 | + lodash-es@4.17.21
11:19:39.980 | + lucide-react@0.359.0
11:19:39.980 | + modern-screenshot@4.4.38
11:19:39.980 | + nanoid@5.0.6
11:19:39.981 | + next@14.1.3
11:19:39.981 | + next-auth@5.0.0-beta.15
11:19:39.981 | + next-sitemap@4.2.3
11:19:39.981 | + numeral@2.0.6
11:19:39.981 | + nuqs@1.17.1
11:19:39.981 | + ollama@0.5.0
11:19:39.981 | + openai@4.29.1
11:19:39.981 | + polished@4.3.1
11:19:39.981 | + posthog-js@1.116.2
11:19:39.981 | + query-string@9.0.0
11:19:39.981 | + react@18.2.0
11:19:39.981 | + react-dom@18.2.0
11:19:39.981 | + react-hotkeys-hook@4.5.0
11:19:39.981 | + react-i18next@14.1.0
11:19:39.981 | + react-layout-kit@1.9.0
11:19:39.981 | + react-lazy-load@4.0.1
11:19:39.981 | + react-virtuoso@4.7.2
11:19:39.981 | + react-wrap-balancer@1.1.0
11:19:39.981 | + remark@14.0.3
11:19:39.981 | + remark-gfm@3.0.1
11:19:39.981 | + remark-html@15.0.2
11:19:39.982 | + rtl-detect@1.1.2
11:19:39.982 | + semver@7.6.0
11:19:39.982 | + sharp@0.33.2
11:19:39.982 | + swr@2.2.5
11:19:39.982 | + systemjs@6.14.3
11:19:39.982 | + ts-md5@1.3.1
11:19:39.982 | + ua-parser-js@1.0.37
11:19:39.982 | + url-join@5.0.0
11:19:39.982 | + use-merge-value@1.2.0
11:19:39.984 | + utility-types@3.11.0
11:19:39.984 | + uuid@9.0.1
11:19:39.984 | + yaml@2.4.1
11:19:39.984 | + zod@3.22.4
11:19:39.985 | + zustand@4.5.2
11:19:39.985 | + zustand-utils@1.3.2
11:19:39.985 |  
11:19:39.985 | 2455 packages installed [36.44s]
11:19:39.985 | warn: tree-sitter's install script took 21.6s
11:19:39.985
```

**Comment by Electroid** [comment]
Duplicate of #9489

**Comment by arvinxx** [comment]
@Electroid @Jarred-Sumner When will this be resolved? If today or tomorrow, I will wait for the new version. Or I have to lock the version to `1.0.31` in my workflow.

**Comment by dylan-conway** [comment]
> @Electroid @Jarred-Sumner When will this be resolved? If today or tomorrow, I will wait for the new version. Or I have to lock the version to `1.0.31` in my workflow.

We will have a fix for this released in `v1.0.34` tonight
